src/map/script.c

```diff
@@ -17900,17 +17895,14 @@ BUILDIN_FUNC(unitattack)
 /// unitstopattack <unit_id>;
 BUILDIN_FUNC(unitstopattack)
 {
-	int unit_id;
 	struct block_list* bl;
 
-	unit_id = script_getnum(st,2);
-	bl = map_id2bl(unit_id);
+	if(!script_rid2bl(2,bl))
+		return SCRIPT_CMD_FAILURE;
```

Comment:
switched from success to failure again. again you could invert the if and put the code back in the if